pandas/core/groupby/indexing.py

```diff
@@ -0,0 +1,227 @@
+from __future__ import annotations
+
+from typing import (
+    Iterable,
+    cast,
+)
+
+import numpy as np
+
+from pandas._typing import (
+    FrameOrSeries,
+    PositionalIndexer,
+)
+from pandas.util._decorators import (
+    cache_readonly,
+    doc,
+)
+
+from pandas.core.dtypes.common import (
+    is_integer,
+    is_list_like,
+)
+
+from pandas.core.groupby import groupby
+from pandas.core.indexes.api import CategoricalIndex
+
+
+class GroupByIndexingMixin:
+    """
+    Mixin for adding .rows to GroupBy.
+    """
+
+    @cache_readonly
+    def _rows(self) -> _rowsGroupByIndexer:
+        return _rowsGroupByIndexer(cast(groupby.GroupBy, self))
+
+
+@doc(GroupByIndexingMixin._rows)
+class _rowsGroupByIndexer:
+    def __init__(self, groupByObject: groupby.GroupBy):
+        self.groupByObject = groupByObject
+
+    def __getitem__(self, arg: PositionalIndexer | tuple) -> FrameOrSeries:
+        """
+        Positional index for selection by integer location per group.
+
+        Used to implement GroupBy._rows which is used to implement GroupBy.nth
+        when keyword dropna is None or absent.
+        The behaviour extends GroupBy.nth and handles DataFrame.groupby()
+        keyword parameters such as as_index and dropna in a compatible way.
+
+        The additions to nth(arg) are:
```

Comment:
this is not desriptive here (e.g. you have to have context to understand), so instead you can just say this (rather than additions to nth)

Reply:
Done